plugins/woocommerce/src/Snapshots/SnapshotsControllerBase.php

```diff
@@ -0,0 +1,313 @@
+<?php
+
+namespace Automattic\WooCommerce\Snapshots;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Base class for snapshot controllers.
+ */
+abstract class SnapshotsControllerBase {
```

Comment:
Since `Snapshots` is in the namespace, does it need to be in the classname as well:

```
\Automattic\WooCommerce\Snapshots\SnapshotsControllerBase
```

vs

```
\Automattic\WooCommerce\Snapshots\ControllerBase
```